pkg/sharing/member.go

```diff
@@ -177,33 +177,53 @@ func (s *Sharing) RevokeMember(inst *instance.Instance, m *Member, c *Credential
 				Warnf("Error on revocation notification: %s", err)
 		}
 
-		if !s.ReadOnly() {
-			if err := DeleteOAuthClient(inst, m, c); err != nil {
-				return err
-			}
+		if err := DeleteOAuthClient(inst, m, c); err != nil {
+			return err
 		}
 	}
 	m.Status = MemberStatusRevoked
-
 	// Do not remove the credential to preserve the members / credentials order
 	*c = Credentials{}
 
 	return couchdb.UpdateDoc(inst, s)
 }
 
+// RevokeOwner revoke the access granted to the owner and notify it
+func (s *Sharing) RevokeOwner(inst *instance.Instance) error {
+	m := &s.Members[0]
+	c := &s.Credentials[0]
+
+	if err := s.NotifyMemberRevocation(inst, m, c); err != nil {
+		inst.Logger().WithField("nspace", "sharing").
+			Warnf("Error on revocation notification: %s", err)
+	}
+	if err := DeleteOAuthClient(inst, m, c); err != nil {
+		return err
+	}
+	m.Status = MemberStatusRevoked
+	s.Credentials = nil
```

Comment:
I would also set `s.Active` to false. What do you think?

Reply:
I actually set it after in `RevokeRecipientBySelf` It follows the same logic than the other revocation functions, eg. revoke the member-related things (notify it, member revoked, crendials reset) and then the sharing-related things (active to false, remove triggers and shared references)